feat: add `math/base/special/factoriallnf`

anandkaranubc opened the pull request:
Progresses https://github.com/stdlib-js/stdlib/issues/649

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the complete implementation for `math/base/special/factoriallnf`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Progresses https://github.com/stdlib-js/stdlib/issues/649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
feat: add `math/base/special/factoriallnf`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/factoriallnf/lib/main.js:
Note: We don't have `isNegativeIntegerf` implemented.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/factoriallnf/lib/main.js:
The other option is to use `!isNonNegativeIntegerf`

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/factoriallnf/lib/main.js:
@anandkaranubc I just added `math/base/assert/is-negative-integerf`, so can rebase this branch / merge in latest changes and then start using `isNegativeIntegerf` here.

anandkaranubc committed:
Merge branch 'develop' into feat/factoriallnf

anandkaranubc committed:
fix: use `isNegativeIntegerf`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/factoriallnf/lib/main.js:
Thanks @Planeshifter!

This is resolved now.

anandkaranubc committed:
build: update `manifest.json`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter reviewed (approved):
Looks great; thanks, @anandkaranubc!
